packages/grid/_modules_/grid/hooks/features/columns/useGridColumns.ts

```diff
@@ -7,149 +7,46 @@ import {
 } from '../../../constants/eventsConstants';
 import { GridApiRef } from '../../../models/api/gridApiRef';
 import { GridColumnApi } from '../../../models/api/gridColumnApi';
-import { gridCheckboxSelectionColDef } from '../../../models/colDef/gridCheckboxSelection';
 import {
   GridColDef,
-  GridColumns,
-  GridColumnsMeta,
   getInitialGridColumnsState,
-  GridInternalColumns,
+  GridColumnsState,
 } from '../../../models/colDef/gridColDef';
-import { GridColumnTypesRecord } from '../../../models/colDef/gridColTypeDef';
-import { getGridDefaultColumnTypes } from '../../../models/colDef/gridDefaultColumnTypes';
-import { getGridColDef } from '../../../models/colDef/getGridColDef';
-import { Logger } from '../../../models/logger';
 import { GridColumnOrderChangeParams } from '../../../models/params/gridColumnOrderChangeParams';
-import { mergeGridColTypes } from '../../../utils/mergeUtils';
 import { useGridApiMethod } from '../../root/useGridApiMethod';
-import { optionsSelector } from '../../utils/optionsSelector';
 import { useLogger } from '../../utils/useLogger';
 import { useGridSelector } from '../core/useGridSelector';
-import { GridLocaleText, GridTranslationKeys } from '../../../models/api/gridLocaleTextApi';
 import { useGridState } from '../core/useGridState';
 import {
   allGridColumnsSelector,
   gridColumnsMetaSelector,
   visibleGridColumnsSelector,
 } from './gridColumnsSelector';
 import { useGridApiOptionHandler } from '../../root/useGridApiEventHandler';
-import { GRID_STRING_COL_DEF } from '../../../models/colDef/gridStringColDef';
 import { GridComponentProps } from '../../../GridComponentProps';
-
-function updateColumnsWidth(columns: GridColumns, viewportWidth: number): GridColumns {
-  const numberOfFluidColumns = columns.filter((column) => !!column.flex && !column.hide).length;
-  let flexDivider = 0;
-
-  if (numberOfFluidColumns && viewportWidth) {
-    columns.forEach((column) => {
-      if (!column.hide) {
-        if (!column.flex) {
-          viewportWidth -= Math.max(column.minWidth!, column.width!);
-        } else {
-          flexDivider += column.flex;
-        }
-      }
-    });
-  }
-
-  let newColumns = columns;
-  if (numberOfFluidColumns) {
-    const flexMultiplier = viewportWidth / flexDivider;
-    newColumns = columns.map((column) => {
-      if (!column.flex) {
-        return {
-          ...column,
-          width: Math.max(column.minWidth!, column.width!),
-        };
-      }
-
-      let newWidth = column.minWidth! ?? GRID_STRING_COL_DEF.width;
-      if (viewportWidth > 0) {
-        const flexColumnWidth = Math.floor(flexMultiplier * column.flex!);
```

Comment:
We forgot this logic, e.g. `min-width: 246.923px;`, if it was made on purpose, I think that we should break the changes down. One problem at a time.

Reply:
There is behavior changes
If you

1) go to this demo : http://localhost:3001/components/data-grid/columns/#resizing
2) set a column flex to 1
3) resize this column
4) resize it again

Between 3 and 4 the column you resize will take bake its original width (that's my video https://github.com/mui-org/material-ui-x/issues/2129#issuecomment-878280780)
But the tests are not testing this particular scenario.

___

OK to split this PR

@oliviertassinari for 4. you see another solution than mine if we want to keep round numbers ?

https://github.com/mui-org/material-ui-x/pull/2104/commits/7b26f5ee3086088ba568a047c13ed5e7ccd22b3c#diff-ab3374e32d059e6ec4d957bf3002146d73129b88296f556ae19061e1c15512c6R70

___

I don't understand what you mean by removing the flex when resizing.
Do you mean just ignore the property if a column has been resized since the last prop column update ?
It would cause jumps when resizing with the selector in the header no ?

Reply:
@flaviendelangle Ok, all good.

1. No strong point of view
2. Yes, it seems fine
3. Yes, but we can always wait to see if it's enough. So it seems that we fix the problem in #2011 here. I guess we could add test cases in another PR and then close #2011.

Reply:
OK, I let this PR on hold and I create the one with the test

Reply:
https://github.com/mui-org/material-ui-x/pull/2211

See this one for the tests